doc/source/workflow.rst

```diff
@@ -0,0 +1,726 @@
+  *PXE* image type.
+
+``premount`` | ``postmount``
+  The hooks are called before and after the client root file system is mounted,
+  respectively. The hooks only exist for the PXE image type. 
+
+``prenetwork`` | ``postnetwork``
+  The hooks are called before and after the client network is setup,
+  respectively. The hooks only exist for the *PXE* image type.
+
+``prepartition`` | ``postpartition``
+  The hooks are called before and after the client creates the partition table
+  on the target disk, respectively. The hooks only exist for the *PXE* image
+  type.
+
+``preprobe`` | ``postprobe``
+  The hooks are called before and after the loading of modules not handled by
+  udev, respectively. The hooks only exist for the *PXE* image type.
+
+``preswap`` | ``postswap``
+  The hooks are called before and after the creation of the swap space,
+  respectively. The hooks only exist for the *PXE* image type.
+
+``preactivate``
+  This hook is called before the root file system is moved to :file:`/`. The
+  hook only exists for the *PXE* image type.
+
+``preCallInit``
+   This hook is called before the initialization process, init or systemd, is
+   started. At call time the root file system has already been moved to
+   :file:`/`. The hook only exists for the *OEM* and *VMX* image types.
+
+``preRecovery`` | ``postRecovery``
+  This hook is called before and after the recovery code is processed. At call
+  time of preRecovery the recovery partition is not yet mounted. At call time
+  of postRecovery the recovery partition is still mounted on :file:`/reco-save`.
+  The hook only exists for the *OEM* image type.
+
+``preRecoverySetup`` | ``postRecoverySetup``
+  This hook is called before and after the recovery setup is processed. At call
+  time of preRecoverySetup the recovery partition is not yet mounted. At call
+  time of postRecoverySetup the recovery partition is still mounted on
+  :file:`/reco-save`. The hook only exists for the *OEM* image type.
+
+``preException``
+  This hook is called before a system error is handled. The error message is
+  passed as parameter. This hook can be used for all image types.
+
+``preHWdetect`` | ``postHWdetect``
+  The hooks are called before and after the install image boot code detects
+  the possible target storage device(s). The hooks only exist for the *OEM*
+  image type.
+  
+``preNetworkRelease``
+  This hook is called before the network connection is released. The hook only
+  exists for the *PXE* image type.
+
+
+Including Hook Scripts into the Boot Image
+''''''''''''''''''''''''''''''''''''''''''
+
+All hook scripts must be located in the :file:`kiwi-hooks` directory at the top
+level of the initrd. The best approach to including the hook scripts in the
+initrd is to create an archive of a :file:`kiwi-hooks` directory that contains
+the custom boot scripts.
+
+.. code-block:: bash
+
+   mkdir kiwi-hooks
+   place all scripts inside kiwi-hooks
```

Comment:
I would precede this line with a hash symbol (= comment), otherwise it could be easily misread as a command (`place`).